File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxAscendingSum = function(nums) {
  let [ans, sum] = [nums[0], nums[0]];
  
  for (let i = 1; i < nums.length; i++) {
    sum = nums[i] > nums[i - 1] ? sum + nums[i] : nums[i];
    ans = Math.max(ans, sum);
  }
  
  return ans;
};

File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        ans, s = nums[0], nums[0]
        for i in range(1, len(nums)):
            if nums[i] > nums[i - 1]:
                s += nums[i]
            else:
                s = nums[i]
            ans = max(ans, s)
        return ans